### DISCUSSION TITLE
Storefront: Adding Videos

### DISCUSSION THREAD

**Discussion by Jicknones** [swellstores/community#559] [Ideas]
(I'm aware that there is officially no support offered for legacy storefronts)

My client would like Videos of their products on product listings, or just on the home page. Anyone have experience adding videos? There is no option to add them built-in, I'm pretty non-technical to be honest. More on the marketing side of things. 

Any ideas or work-arounds would be greatly appreciated. Thanks a ton!

**Comment by sags95** [comment]
Hi @Jicknones, this would definitely be possible but you would just have to find a way to host the product videos. There are many services that offer this such as Cloudinary, ImageKit, etc. Once the assets are hosted there you can then link the URLs on the products through a custom field.

Custom fields can be easily created through the dashboard, adding a link to a guide [here](https://developers.swell.is/guides/custom-fields-and-models#creating-custom-fields). You can disregard the other steps as you would just need a text field to add the video URLs. The most important step would be to ensure you're selecting "content" under the "Advanced Options" section, this ensures that the field will be included with the Frontend API.

Once the video urls are set on the products you should then be able to render them on the storefront.
